osquery/tables/system/darwin/cups_destinations.cpp

```diff
@@ -0,0 +1,59 @@
+/**
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under both the Apache 2.0 license (found in the
+ *  LICENSE file in the root directory of this source tree) and the GPLv2 (found
+ *  in the COPYING file in the root directory of this source tree).
+ *  You may select, at your option, one of the above-listed licenses.
+ */
+
+#include <cups/cups.h>
+
+#include <osquery/system.h>
+#include <osquery/tables.h>
+
+namespace osquery {
+namespace tables {
+
+class SafeCupsDestinations {
+ public:
+  cups_dest_t* destination_list;
+  int num_destinations;
+
+  SafeCupsDestinations() {
+    num_destinations = cupsGetDests(&destination_list);
+  }
+
+  ~SafeCupsDestinations() {
+    cupsFreeDests(num_destinations, destination_list);
+  }
+};
+
+QueryData genCupsDestinations(QueryContext& request) {
+  QueryData results;
+  SafeCupsDestinations destinations;
+
+  for (auto i{0}; i < destinations.num_destinations; ++i) {
```

Comment:
Is there a reason not to use `int i = 0` here and the other loops added?

Reply:
Not particularly, I've just been trying to use braced initialization and auto as much as possible. I suppose the more correct thing would be to use `decltype(destinations.num_destination)` instead of auto though.

Reply:
I believe `decltype` is the correct tool for this job as the for loop iterator would change it's type correctly if the class member `num_destinations` ever changed type. This seems to be right thing to do from code correctness standpoint.

I do agree that in this particular usage is somewhat contrived but I like the idea of apply proper software engineer principals across the codebase no matter where they are.

Does `decltype` give you the same issue as `auto` in this case? I don't believe this would be mistaken for a range based loop but I can see how that could happen in the future if one of the other clauses where omitted. If so, maybe we can move that discussion to the slack?